bundles/org.openhab.binding.modbus.stiebeleltron/src/main/java/org/openhab/binding/modbus/stiebeleltron/internal/handler/StiebelEltronHandler.java

```diff
@@ -0,0 +1,744 @@
+
+    }
+
+    /**
+     * This method is called each time new data has been polled from the modbus
+     * slave The register array is first parsed, then each of the channels are
+     * updated to the new values
+     *
+     * @param registers byte array read from the modbus slave
+     */
+    protected void handlePolledSystemParameterData(ModbusRegisterArray registers) {
+        logger.trace("System state block received, size: {}", registers.size());
+
+        SystemParameterBlock block = systemParameterBlockParser.parse(registers);
+        updateState(channelUID(GROUP_SYSTEM_PARAMETER, CHANNEL_OPERATION_MODE), new DecimalType(block.operation_mode));
+
+        updateState(channelUID(GROUP_SYSTEM_PARAMETER, CHANNEL_COMFORT_TEMPERATURE_HEATING),
+                getScaled(block.comfort_temperature_heating, CELSIUS));
+        updateState(channelUID(GROUP_SYSTEM_PARAMETER, CHANNEL_ECO_TEMPERATURE_HEATING),
+                getScaled(block.eco_temperature_heating, CELSIUS));
+        updateState(channelUID(GROUP_SYSTEM_PARAMETER, CHANNEL_COMFORT_TEMPERATURE_WATER),
+                getScaled(block.comfort_temperature_water, CELSIUS));
+        updateState(channelUID(GROUP_SYSTEM_PARAMETER, CHANNEL_ECO_TEMPERATURE_WATER),
+                getScaled(block.eco_temperature_water, CELSIUS));
+    }
+
+    /**
+     * @param bridgeStatusInfo
+     */
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        super.bridgeStatusChanged(bridgeStatusInfo);
+
+        logger.debug("Thing status changed to {}", this.getThing().getStatus().name());
+        if (getThing().getStatus() == ThingStatus.ONLINE) {
+            startUp();
+        } else if (getThing().getStatus() == ThingStatus.OFFLINE) {
+            tearDown();
+        }
+    }
+
+    /**
+     * Handle errors received during communication
+     */
+    protected void handleError(@Nullable Exception error) {
+        // Ignore all incoming data and errors if configuration is not correct
+        if (hasConfigurationError() || getThing().getStatus() == ThingStatus.OFFLINE) {
+            return;
+        }
+        String msg = "";
+        String cls = "";
+        if (error != null) {
+            cls = error.getClass().getName();
+            msg = error.getMessage();
+        }
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                String.format("Error with read: %s: %s", cls, msg));
+    }
+
+    /**
+     * Returns true, if we're in a CONFIGURATION_ERROR state
+     *
+     * @return
+     */
+    protected boolean hasConfigurationError() {
+        ThingStatusInfo statusInfo = getThing().getStatusInfo();
+        return statusInfo.getStatus() == ThingStatus.OFFLINE
+                && statusInfo.getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR;
+    }
+
+    /**
+     * Reset communication status to ONLINE if we're in an OFFLINE state
+     */
+    protected void resetCommunicationError() {
```

Comment:
Is this method used anywhere?

Reply:
I have inserted the method call in the polling data handlers